imagetest/testworkflow.go

```diff
@@ -0,0 +1,245 @@
+package imagetest
+
+import (
+	"context"
+	"encoding/xml"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+	"sync"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"github.com/GoogleCloudPlatform/compute-image-tools/daisy"
+	"github.com/GoogleCloudPlatform/guest-test-infra/imagetest/utils"
+)
+
+var (
+	client *storage.Client
+)
+
+const (
+	testBinariesPath  = "/out"
+	testWrapperPath   = testBinariesPath + "/wrapper"
+	baseGCSPath       = "gs://gcp-guest-test-outputs/cloud_image_tests/"
+	createVMsStepName = "create-vms"
+)
+
+// TestWorkflow defines a test workflow which creates at least one test VM.
+type TestWorkflow struct {
+	Image          string
+	Name           string
+	ShortImage     string
+	destination    string
+	skipped        bool
+	skippedMessage string
+	wf             *daisy.Workflow
+}
+
+func finalizeWorkflows(tests []*TestWorkflow, zone, project string) error {
```

Comment:
Since the logic is fully within the for loop below, finalizeWorkflow (singular) could naturally be a method on TestWorkflow struct. You can keep finalizeWorkflows if that makes code shorter elsewhere.

Also, add docs what this does in summary since it's not clear without going through the whole method. Just looking at the name, it sounded like this is wrapping up tests (once they finished), but it's actually building/extending test Daisy workflows. It's more of a builder function.

Reply:
yes it 'finalizes' a workflow to make it ready to run. will look at updating as you suggest